arches/app/etl_modules/import_single_csv.py

```diff
@@ -72,13 +96,57 @@ def get_node_lookup(self, graphid):
             self.node_lookup[graphid] = Node.objects.filter(graph_id=graphid)
         return self.node_lookup[graphid]
 
-    def read(self, request):
+    def cli(self, source):
+        initiated = self.start(self.request)
+
+        if initiated["success"]:
+            read = self.read(source=source)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": initiated["message"]},
+            }
+
+        if read["success"]:
+            written = self.write(self.request)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": read["message"]},
+            }
+
+        if written["success"]:
+            return {"success": True, "data": "done"}
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": written["message"]},
+            }
```

Comment:
`read` and `written` shouldn't be defined in `if` statements because this could fail with undefined errors.

Reply:
The variables are now defined beforehand. The unexpected errors are handled, too.